std/math.d

```diff
@@ -464,24 +463,42 @@ T floorImpl(T)(const T x) @trusted pure nothrow @nogc
             return 0.0;
     }
 
-    exp = (T.mant_dig - 1) - exp;
-
-    // Zero 16 bits at a time.
-    while (exp >= 16)
+    static if (F.realFormat == RealFormat.ieeeSingle)
     {
-        version (LittleEndian)
-            y.vu[pos++] = 0;
-        else
-            y.vu[pos--] = 0;
-        exp -= 16;
+        if (exp < (T.mant_dig - 1))
+        {
+            const uint i = F.MANTISSAMASK_INT >> exp;
+
+            // Clear all bits representing the fraction part.
+            if ((y.vi & i) != 0)
+            {
+                if (y.vi < 0)
+                    y.vi += 0x00800000 >> exp;
```

Comment:
This REALLY needs a comment. My guess is that this is the float value 0.5, added so that it rounds up towards zero? Or something like that.

Reply:
This is subtracting one, adding comment.